### PR TITLE
Add integration tests for id token encryption

### PR DESCRIPTION
**Pull request by vihanga-liyanage**
Integration test added to test ID token encryption feature in following grant types.

- Authorization Code Grant
- Implicit Grant
- Resource Owner Credential Grant

This will fix [wso2/product-is#2460](https://github.com/wso2/product-is/issues/2460)

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java [hunk 1]
```diff
@@ -481,4 +495,109 @@ public String requestAccessToken(String consumerKey, String consumerSecret,
     public String getBase64EncodedString(String consumerKey, String consumerSecret) {
         return new String(Base64.encodeBase64((consumerKey + ":" + consumerSecret).getBytes()));
     }
+
+	/**
+	 * Convert a x509 certificate to pem format.
+	 * @param x509Certificate	Certificate in x509 format.
+	 * @return					Certificate in pom format.
```
Comment (chirankavinda123) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java]: I guess it should be pem, not pom

Comment (vihanga-liyanage) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java]: Ack.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java [hunk 2]
```diff
@@ -481,4 +495,109 @@ public String requestAccessToken(String consumerKey, String consumerSecret,
...
+	 * @param publicKey		RSAPublicKey for the certificate.
+	 * @param privateKey	RSAPrivateKey for the certificate.
+	 * @return
+	 * @throws CertificateEncodingException
+	 * @throws NoSuchAlgorithmException
+	 * @throws SignatureException
+	 * @throws InvalidKeyException
+	 */
+	public X509Certificate getX509PublicCert(RSAPublicKey publicKey, RSAPrivateKey privateKey)
+			throws CertificateEncodingException, NoSuchAlgorithmException, SignatureException, InvalidKeyException {
```
Comment (menakaj) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java]: Shall we obey the 120 column limit here?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/utils/DataExtractUtil.java [hunk 3]
```diff
@@ -237,6 +243,45 @@ public static String extractAccessTokenFromQueryString(String query) {
...
+	 *
+	 * @param URIString		URI as a string   .
+	 * @param key			Key that needs to be extracted.
+	 * @return				String value corresponds to the key.
+	 * @throws URISyntaxException
+	 */
+	public static String getParamFromURIString(String URIString, String key) throws URISyntaxException {
+
+		String param = null;
+		List<NameValuePair> params = URLEncodedUtils.parse(new URI(URIString), String.valueOf(StandardCharsets.UTF_8));
```
Comment (menakaj) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/utils/DataExtractUtil.java]: Fix here as well.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java (hunk 2 above)
Comment (vihanga-liyanage) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java]: It ends at 117. :-)

File: modules/integration/tests-integration/tests-backend/pom.xml [hunk 4]
```diff
@@ -648,5 +648,11 @@
             <groupId>com.nimbusds</groupId>
             <artifactId>nimbus-jose-jwt</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.nimbusds</groupId>
+            <artifactId>oauth2-oidc-sdk</artifactId>
+            <version>5.52</version>
```
Comment (omindu) [inline modules/integration/tests-integration/tests-backend/pom.xml]: Move to parent pom

**Pull request merged**